src/lib.rs

```diff
@@ -644,6 +645,40 @@ where
     }
 }
 
+/// A panic payload indicating that a salsa revision was canceled.
+#[derive(Debug)]
+pub struct Canceled {
+    _private: (),
+}
+
+impl Canceled {
+    fn throw() -> ! {
+        std::panic::resume_unwind(Box::new(Self { _private: () }));
+    }
+}
+
+impl std::fmt::Display for Canceled {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str("canceled")
+    }
+}
+
+impl std::error::Error for Canceled {}
+
+/// Runs `f`, and catches any salsa cancelation.
+pub fn catch_cancellation<F, T>(f: F) -> Result<T, Canceled>
```

Comment:
Maybe `Cancelled::catch`? I am a huge fan of exporting a single name and letting the rest of the API hang off it.